src/snowflake/snowpark/session.py

```diff
@@ -191,6 +191,38 @@ class Session:
     A :class:`Session` object is not thread-safe.
     """
 
+    class RuntimeConfig:
+        def __init__(self, session: "Session", conf: Dict[str, Any]) -> None:
+            self._session = session
+            for key, val in conf.items():
+                if hasattr(Session, key):
+                    setattr(session, key, val)
+
+        def get(self, key: str, default=None) -> Any:
+            if hasattr(Session, key):
+                return getattr(self._session, key)
+            if hasattr(self._session._conn._conn, key):
+                return getattr(self._session._conn._conn, key)
+            return default
+
+        def is_modifiable(self, key: str) -> bool:
+            if hasattr(Session, key) and isinstance(getattr(Session, key), property):
+                return getattr(Session, key).fset is not None
+            if hasattr(SnowflakeConnection, key) and isinstance(
+                getattr(SnowflakeConnection, key), property
+            ):
+                return getattr(SnowflakeConnection, key).fset is not None
+            return False
+
+        def set(self, key: str, value: Any) -> None:
+            if self.is_modifiable(key):
+                if hasattr(Session, key):
+                    setattr(self._session, key, value)
+                if hasattr(SnowflakeConnection, key):
+                    setattr(self._session._conn._conn, key, value)
```

Comment:
Is it expected that the `key` exists in both `Session` and `SnowflakeConnection`? Also, if neither class has the key, this becomes a no-op. Just want to make sure if this is the expected behavior.

Reply:
`telemetry_enabled` is the only property that exists in both `Session` and `SnowflakeConnection` for now. Since `is_modifiable` returns `true`, this key must exist in one of the classes.